Sources/Core/HardwareBreakpointManager.cpp

```diff
@@ -40,9 +39,12 @@ ErrorCode HardwareBreakpointManager::add(Address const &address, Type type,
 }
 
 ErrorCode HardwareBreakpointManager::remove(Address const &address) {
-  auto loc = std::find(_locations.begin(), _locations.end(), address);
+  auto loc = std::find_if(_locations.begin(), _locations.end(),
+                          [address](const Location &l){
+                            return l.address == address;
+                          });
```

Comment:
Actually, I have a different idea. We could just use a std::map for `_locations`, and remove the `DS2ASSERT(_locations.size() == maxWatchpoints());` (since you can't fix the size of a map). we already check `_sites.size() >= maxWatchpoints()` in `HardwareBreakpointManager::add`, so there's really no reason to also restrict the size of `_locations`. If you use <address, value> as <key, value> in a map, you don't need the find_if's anymore.